tron/core/job_scheduler.py

```diff
@@ -73,9 +80,17 @@ def manual_start(self, run_time=None):
     def schedule_reconfigured(self):
         """Remove the pending run and create new runs with the new JobScheduler.
         """
-        if self.job.enabled:
-            self.job.runs.remove_pending()
-            self.create_and_schedule_runs(ignore_last_run_time=True)
+        if not self.job.enabled:
+            return
+
+        # when reconfiguring, preserve the latest scheduled run's time
+        pending_run_times = [j.run_time for j in list(self.job.runs.get_pending())]
+        if len(pending_run_times) != 1:
+            log.warning(f'{self.job} has {len(pending_run_times)} pending runs, not 1')
+        next_run_time = 'current' if len(pending_run_times) == 0 else pending_run_times[0]
+
+        self.job.runs.remove_pending()
```

Comment:
Maybe this could return the runs it removes?

Reply:
That wasn't the case before the changes. Why do you want this?

Reply:
Oh, I was suggesting you could use the run time from the removed pending runs, instead of having to find it separately.